test/hotspot/jtreg/serviceability/sa/sadebugd/PmapOnDebugdTest.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) 2021, Oracle and/or its affiliates. All rights reserved.
+ * Copyright (c) 2021 NTT DATA.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import jdk.test.lib.JDKToolLauncher;
+import jdk.test.lib.apps.LingeredApp;
+import jdk.test.lib.process.OutputAnalyzer;
+import jdk.test.lib.SA.SATestUtils;
+
+import jtreg.SkippedException;
+
+/**
+ * @test
+ * @bug 8263670
+ * @requires vm.hasSA
+ * @requires os.family != "windows"
+ * @library /test/lib
+ * @run main/othervm PmapOnDebugdTest
+ */
+
+public class PmapOnDebugdTest {
+
+    public static void main(String[] args) throws Exception {
+        SATestUtils.skipIfCannotAttach(); // throws SkippedException if attach not expected to work.
+
+        if (SATestUtils.needsPrivileges()) {
+            // This tests has issues if you try adding privileges on OSX. The debugd process cannot
+            // be killed if you do this (because it is a root process and the test is not), so the destroy()
+            // call fails to do anything, and then waitFor() will time out. If you try to manually kill it with
+            // a "sudo kill" command, that seems to work, but then leaves the LingeredApp it was
+            // attached to in a stuck state for some unknown reason, causing the stopApp() call
+            // to timeout. For that reason we don't run this test when privileges are needed. Note
+            // it does appear to run fine as root, so we still allow it to run on OSX when privileges
+            // are not required.
+            throw new SkippedException("Cannot run this test on OSX if adding privileges is required.");
+        }
```

Comment:
pmap and pstack are not even supported on OSX with a live process. The automated github testing did not catch this because it passed due to the above check:

TEST RESULT: Passed. Skipped: jtreg.SkippedException: Cannot run this test on OSX if adding privileges is required.

If it passed the check, it would have failed below to find a .dylib. You could change the code below to something like what is in ClhsdbPmap.java:

```
            if (!withCore && Platform.isOSX()) {
                expStrMap.put("pmap", List.of("Not available for Mac OS X processes"));
            } else {
                expStrMap.put("pmap", List.of("jvm", "java", "jli", "jimage"));
            }
```